### ISSUE TITLE
Somatic pipeline bugs

### ISSUE THREAD

**Issue by ericblanc20** [bihealth/snappy-pipeline#143]
# List of bugs

There bugs were encountered while running an end-to-end example on Becnel data.

## Memory allocation

Errors found in `__init__.py` for workflows:

- `cbioportal_export`: lines [306-310](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_pipeline/workflows/cbioportal_export/__init__.py#L306-L310)
- `gene_expression_quantification`: multiples instances (lines [324-328](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_pipeline/workflows/gene_expression_quantification/__init__.py#L324-L328), [348-352](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_pipeline/workflows/gene_expression_quantification/__init__.py#L348-L352) and [360-364](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_pipeline/workflows/gene_expression_quantification/__init__.py#L360-L364))
- `panel_of_normals`: lines [339-348](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_pipeline/workflows/panel_of_normals/__init__.py#L339-L348)
- `somatic_variant_calling`: multiple instances (lines [744-768](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_pipeline/workflows/somatic_variant_calling/__init__.py#L744-L768) & [801-805](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_pipeline/workflows/somatic_variant_calling/__init__.py#L801-L805))

## Default configuration

Errors found in `__init__.py` for workflows:

- `cbioportal_export`: duplicate parameters on lines [66](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_pipeline/workflows/cbioportal_export/__init__.py#L66) & [68-70](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_pipeline/workflows/cbioportal_export/__init__.py#L68-L70).

## Location of executables

Errors found in wrappers

- `rnaqc/rnaseqc/wrapper.py`: the hard-coded jar version on [line 70](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_wrappers/wrappers/rnaqc/rnaseqc/wrapper.py#L70) causes an error. Suggestion for change: `jar_path=$(find ${{JAVA_HOME}}/share -name 'RNA-SeQC_v*.jar')``
- `snappy_wrappers/wrappers/eb_filter/wrapper.py`: the snappy script `snappy-vcf_sort` is not found on [line 86](https://github.com/bihealth/snappy-pipeline/blob/d50b85dce2d871e0384cd35a7f64dbc2bff8b28f/snappy_wrappers/wrappers/eb_filter/wrapper.py#L86). Suggestion for change: `import os` and `snappy_vcf_sort = os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(os.path.realpath(__file__))))), "scripts", "snappy-vcf_sort")`in the python part, and replacing line 86 with `| {snappy_vcf_sort} $REF.fai`.

## Environment

Errors found in environment definitions of the wrappers

- `eb_filter`: added `gsl==2.7`, and fixed version of `bcftools==1.15.1` & `samtools==1.15.1`.
- `eb_filter_par`: added `gsl==2.7`, and fixed version of `bcftools==1.15.1` & `samtools==1.15.1`. `snakemake` was left unpinned. `r` & `http://conda.cubi.bihealth.org/cubiconda`were removed from the list of channels.
- `fastqc`: added `perl ==5.32.1` & `conda-forge` channel, pinned `fastqc==0.11.9` & removed `r` channel.
- `jannovar`: pinned `jannovar-cli==0.36`
- `somatic_variant_filtration/apply_filters`: added `gsl==2.7`, and fixed version of `bcftools==1.15.1` & `htslib==1.15.1`.

**Comment by eudesbarbosa** [comment]
Hello, @ericblanc20, thank you for reporting the issues!

**Memory Allocation:** All `update_cluster` methods will be replaced. So this particular issue will disappear when we merge #141.
**Default configuration:** Issue addressed in #127, will be definitely closed once we merge #141. 
**Location of executables:** 
* `rnaqc/rnaseqc/wrapper.py`: This can probably be an individual Issue. It should be straightforward to fix with no conflicts to the bigger DRMAA removal project.
* `snappy_wrappers/wrappers/eb_filter/wrapper.py`: There was a similar issue in the Freebayes wrapper that I decided to solve by reintroducing an old hack - see #136.

**Environment:** All `gsl` dependency errors were solved in #131. The remaining issues could benefit from individual issues and most likely be fixed in parallel to the DRMAA project.
